src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -87,13 +88,40 @@ public class AsyncCursor<TDocument> : IAsyncCursor<TDocument>
             IBsonSerializer<TDocument> serializer,
             MessageEncoderSettings messageEncoderSettings,
             TimeSpan? maxTime = null)
+            : this(
+                channelSource,
+                collectionNamespace,
+                query,
+                firstBatch,
+                cursorId,
+                null, // postBatchResumeToken
+                batchSize,
+                limit,
+                serializer,
+                messageEncoderSettings)
+        {
+        }
+
+        internal AsyncCursor(
```

Comment:
The other constructor is public, so this one might as well be public also.

Reply:
Done